source/utilities.cc

```diff
@@ -938,12 +948,166 @@ namespace aspect
                                      const MPI_Comm &comm)
     {
       std::string data_string;
+      std::stringstream urlString;
```

Comment:
move this declaration into the scope and as close to the first place you use it.